jsapp/js/account/usage/yourPlan.component.tsx

```diff
@@ -1,25 +1,47 @@
-import usageStyles from 'js/account/usage/usage.module.scss';
 import styles from 'js/account/usage/yourPlan.module.scss';
-import React, {useContext, useMemo, useState} from 'react';
-import {formatDate} from 'js/utils';
-import Badge from 'js/components/common/badge';
+import React, {useEffect, useMemo, useState} from 'react';
+import {formatDate, unixToUtc} from 'js/utils';
+import Badge, {BadgeColor} from 'js/components/common/badge';
 import subscriptionStore from 'js/account/subscriptionStore';
 import sessionStore from 'js/stores/session';
 import BillingButton from 'js/account/plans/billingButton.component';
 import {ACCOUNT_ROUTES} from 'js/account/routes';
 import envStore from 'js/envStore';
-import {PlanNames} from 'js/account/stripe.types';
-import {UsageContext} from 'js/account/usage/useUsage.hook';
+import {PlanNames, BaseProduct, Product} from 'js/account/stripe.types';
+import {getProducts} from '../stripe.api';
 import LimitNotifications from 'js/components/usageLimits/limitNotifications.component';
 
+type subscriptiontype =
+  | 'cancellation'
+  | 'renewal'
+  | 'productChange'
+  | 'priceChange'
+  | '';
```

Comment:
There are a lot of string comparisons to these values; the code would be less vulnerable to accidental typos (e.g. `cancelation`) if you turned this into an `enum` and then used it to test/assign values.